CalibMuon/CSCCalibration/plugins/CSCL1TPLookupTableEP.cc

```diff
@@ -0,0 +1,289 @@
+                                          edm::ValidityInterval& oValidity) {
+  edm::ValidityInterval infinity(iosv.beginOfTime(), iosv.endOfTime());
+  oValidity = infinity;
+}
+
+std::unique_ptr<CSCL1TPLookupTableCCLUT> CSCL1TPLookupTableEP::produceCCLUT(const CSCL1TPLookupTableCCLUTRcd&) {
+  // make the LUT object
+  std::unique_ptr<CSCL1TPLookupTableCCLUT> lut = std::make_unique<CSCL1TPLookupTableCCLUT>();
+
+  // get the text files
+  std::vector<std::string> positionLUTFiles_ = pset_.getParameter<std::vector<std::string>>("positionLUTFiles");
+  std::vector<std::string> slopeLUTFiles_ = pset_.getParameter<std::vector<std::string>>("slopeLUTFiles");
+
+  std::unordered_map<unsigned, std::vector<unsigned>> cclutPosition;
+  std::unordered_map<unsigned, std::vector<unsigned>> cclutSlope;
+
+  // read the text files and extract the data
+  for (int i = 0; i < 5; ++i) {
+    cclutPosition[i] = load(positionLUTFiles_[i]);
+    cclutSlope[i] = load(slopeLUTFiles_[i]);
+  }
+
+  // set the data in the LUT object
+  lut->set_cclutPosition(std::move(cclutPosition));
+  lut->set_cclutSlope(std::move(cclutSlope));
+
+  return lut;
+}
+
+std::unique_ptr<CSCL1TPLookupTableME11ILT> CSCL1TPLookupTableEP::produceME11ILT(const CSCL1TPLookupTableME11ILTRcd&) {
+  // make the LUT object
+  std::unique_ptr<CSCL1TPLookupTableME11ILT> lut = std::make_unique<CSCL1TPLookupTableME11ILT>();
+
+  // get the text files
+  std::vector<std::string> padToHsME1aFiles_ = pset_.getParameter<std::vector<std::string>>("padToHsME1aFiles");
+  std::vector<std::string> padToHsME1bFiles_ = pset_.getParameter<std::vector<std::string>>("padToHsME1bFiles");
+
+  std::vector<std::string> padToEsME1aFiles_ = pset_.getParameter<std::vector<std::string>>("padToEsME1aFiles");
+  std::vector<std::string> padToEsME1bFiles_ = pset_.getParameter<std::vector<std::string>>("padToEsME1bFiles");
+
+  std::vector<std::string> rollToMaxWgME11Files_ = pset_.getParameter<std::vector<std::string>>("rollToMaxWgME11Files");
+  std::vector<std::string> rollToMinWgME11Files_ = pset_.getParameter<std::vector<std::string>>("rollToMinWgME11Files");
+
+  std::vector<std::string> gemCscSlopeCosiFiles_ = pset_.getParameter<std::vector<std::string>>("gemCscSlopeCosiFiles");
+  std::vector<std::string> gemCscSlopeCosiCorrectionFiles_ =
+      pset_.getParameter<std::vector<std::string>>("gemCscSlopeCosiCorrectionFiles");
+  std::vector<std::string> gemCscSlopeCorrectionFiles_ =
+      pset_.getParameter<std::vector<std::string>>("gemCscSlopeCorrectionFiles");
+
+  std::vector<std::string> esDiffToSlopeME1aFiles_ =
+      pset_.getParameter<std::vector<std::string>>("esDiffToSlopeME1aFiles");
+  std::vector<std::string> esDiffToSlopeME1bFiles_ =
+      pset_.getParameter<std::vector<std::string>>("esDiffToSlopeME1bFiles");
+
+  // read the text files and extract the data
+  const auto& GEM_pad_CSC_hs_ME1a_even_ = load(padToHsME1aFiles_[0]);
```

Comment:
Change the type from `const auto&` to just `auto`. This will allow the `std::move` to work.

Reply:
Very good!